src/subgraph/spec.rs

```diff
@@ -643,8 +643,7 @@ impl LinkSpecDefinitions {
                 InputValueDefinition {
                     description: None,
                     name: name!("url"),
-                    // TODO: doc-comment disagrees with non-null here
-                    ty: ty!(String!).into(),
+                    ty: ty!(String).into(),
```

Comment:
this is the definition for `url`, which by spec is `String!`

Reply:
guessing the confusion is from the comment on the method?

```
    ///   directive @link(url: String, as: String, import: [Import], for: link__Purpose) repeatable on SCHEMA
    pub fn link_directive_definition(&self) -> Result<DirectiveDefinition, FederationSpecError> {
```

Yeah it should be `url: String!` as it is a required param -> https://specs.apollo.dev/link/v1.0/